src/bcrypt/__init__.py

```diff
@@ -39,18 +39,19 @@ def gensalt(rounds=12, prefix=b"2b"):
     if prefix not in (b"2a", b"2b"):
         raise ValueError("Supported prefixes are b'2a' or b'2b'")
 
+    # TODO: is this the right min/max?
+    if rounds < 4 or rounds > 31:
+        raise ValueError("Invalid rounds")
+
     salt = os.urandom(16)
     output = _bcrypt.ffi.new("unsigned char[]", 30)
+    _bcrypt.lib.encode_base64(output, salt, len(salt))
 
-    retval = _bcrypt.lib.crypt_gensalt_rn(
-        b"$" + prefix + b"$", rounds, salt, len(salt), output, len(output),
+    return (
+        b"$" + prefix + b"$" + ("%2.2u" % rounds).encode("ascii") + b"$" +
```

Comment:
Reviewing history, I did actually check it properly: `%` works with `bytes` as of 3.5. (Seeing as 3.3 is the compatibility target, though, that doesn’t really matter.)